### PR TITLE
 netinfo: Migrate to community NetInfo v3.2.1.

### LINKED ISSUE #3568: Migrate to community NetInfo
This is a followup to the RN v0.59 upgrade #3399, and a blocker for the RN v0.60 upgrade #3548.

When launching the app with the debugger, the following message now goes to the console:

> Warning: NetInfo has been extracted from react-native core and will be removed in a future release. It can now be installed and imported from '@react-native-community/netinfo' instead of 'react-native'. See https://github.com/react-native-community/react-native-netinfo

In v0.60, the core module appears to indeed be gone.

The README says the new module's API is compatible, so this may be easy.

### PR DESCRIPTION
**Pull request by jainkuniya**
Fixes #3568.

### CODE REVIEW

File: android/app/src/main/java/com/zulipmobile/MainApplication.java [hunk 1]
```diff
@@ -38,6 +39,7 @@ public boolean getUseDeveloperSupport() {
         protected List<ReactPackage> getPackages() {
             return Arrays.asList(
                     new MainReactPackage(),
+            new NetInfoPackage(),
```
Comment (gnprice) [inline android/app/src/main/java/com/zulipmobile/MainApplication.java]: nit: match indentation

File: package.json [hunk 2]
```diff
@@ -32,6 +32,7 @@
   },
   "dependencies": {
     "@expo/react-native-action-sheet": "^1.1.0",
+    "@react-native-community/netinfo": "3.2.1",
```
Comment (gnprice) [inline package.json]: Please use a leading `^` -- that way a plain `yarn upgrade` in the future will get the latest 3.x.

(Notice that almost all our dependencies use that style.)

File: flow-typed/npm/react-native-document-picker_vx.x.x.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+// flow-typed signature: e70237f3f0f0f7216ac66fa336d6aecf
+// flow-typed version: <<STUB>>/react-native-document-picker_vgit://github.com/Elyx0/react-native-document-picker.git#v3/flow_v0.92.0
```
Comment (gnprice) [inline flow-typed/npm/react-native-document-picker_vx.x.x.js]: flow types: Add type for react-native-document-picker.
    
    This dependency was added in d9bbb5e5f047a6de17eef1870fbad17c8debf5f9
    but flow type were not added.
    
    So now just run `flow-typed update` and it is generated automatically.

Heh, we actually do have types for this! Look in the parent directory, and then at the history :slightly_smiling_face: 

Or most practically: go to `ComposeMenu.js` where we use this library, in the `handleFilePicker` function, and try editing `response.name` or something to misspell the property's name. You'll get a Flow error.

I did leave things a little messy in that `yarn update-libdefs` creates this file -- really we want to leave things in a state where it makes no changes unless there are updates to be had. So thanks for tidying that up!

I'll take a few minutes to see if I can teach `yarn update-libdefs` to ignore this library, and if not then I'll merge this commit.

Comment (gnprice) [reply flow-typed/npm/react-native-document-picker_vx.x.x.js]: Merged, as 6219e5a23 . Thanks again!

**Comment by gnprice** [comment]
(BTW I just edited the PR description to point at the issue. The reference in the commit message is important, but a reference in the PR description is helpful to for cross-linking them here on the GitHub web UI.)

**Comment by gnprice** [comment]
Oh, one more remark on the commit message:

    Not upgrading to latest 4.x.x because it comes with AndroidX, which will
    not work for us yet.

Dang, and that is good to know!

When there's a "yet" like this, I always want to answer the question: "until when?" "what's it blocked on?" In this case, I think the answer is the RN v0.60 upgrade: #3548 . Does that agree with your understanding? If so, a link to that issue would be a good way to answer the question, like: "which will not work for us until we upgrade to RN v0.60 (#3548)".

File: package.json (hunk 2 above)
Comment (jainkuniya) [reply package.json]: Yup!

But I thought, there will not be any new 3.x.x release, as they already moved to 4.x.x

Anyway I agree it to have `^`, doing it 👍

**Comment by gnprice** [comment]
Looks good, merged -- thanks @jainkuniya !

**Pull request merged**